p2p/switch.go

```diff
@@ -231,11 +239,15 @@ func (sw *Switch) OnStop() {
 // NOTE: This performs a blocking handshake before the peer is added.
 // NOTE: If error is returned, caller is responsible for calling peer.CloseConn()
 func (sw *Switch) addPeer(peer *peer) error {
+	// Avoid self
+	if sw.nodeKey.ID() == peer.ID() {
```

Comment:
Me likey!
Although perhaps let's make this a sentinel error that can be checked later on in code e.g
```go
var ErrSelfConnection = errors.New("Ignoring connection from self"

...

switch sw.AddPeer(p) {
case nil:
  // Cool
case ErrSelfConnection:
  // Not severe, so we can ignore
default:
  // Handle this error
}
```